component/src/main/java/org/wso2/extension/siddhi/gpl/execution/streamingml/util/CoreUtils.java

```diff
@@ -0,0 +1,184 @@
+                if (!isNumeric(featureAttributeType)) {
+                    throw new SiddhiAppValidationException("model.features in " + (i + 1) + "th parameter is not " +
+                            "a numerical type attribute. Found " +
+                            attributeExpressionExecutors[i].getReturnType()
+                            + ". Check the input stream definition.");
+                }
+
+            } else {
+                throw new SiddhiAppValidationException((i + 1) + "th parameter is not " +
+                        "an attribute (VariableExpressionExecutor) present in the stream definition. Found a "
+                        + attributeExpressionExecutors[i].getClass().getCanonicalName());
+            }
+        }
+        return featureVariableExpressionExecutors;
+    }
+
+
+    public static boolean isNumeric(Attribute.Type attributeType) {
+        return numericTypes.contains(attributeType);
+    }
+
+    public static boolean isLabelType(Attribute.Type attributeType) {
+        return labelTypes.contains(attributeType);
+    }
+
+
+    /**
+     * @param inputDefinition
+     * @param attributeExpressionExecutors
+     * @param classIndex
+     * @return
+     */
+    public static VariableExpressionExecutor extractAndValidateClassLabel
+    (AbstractDefinition inputDefinition, ExpressionExecutor[] attributeExpressionExecutors, int classIndex) {
+        VariableExpressionExecutor classLabelVariableExecutor;
+
+        if (attributeExpressionExecutors[classIndex] instanceof VariableExpressionExecutor) {
+            // other attributes should be numeric type.
+            String attributeName = ((VariableExpressionExecutor)
+                    attributeExpressionExecutors[classIndex]).getAttribute().getName();
+            Attribute.Type classLabelAttributeType = inputDefinition.
+                    getAttributeType(attributeName);
+
+            //class label should be String or Bool
+            if (isLabelType(classLabelAttributeType)) {
+                classLabelVariableExecutor = (VariableExpressionExecutor) attributeExpressionExecutors[classIndex];
+            } else {
+                throw new SiddhiAppValidationException(String.format("[label attribute] in %s th index of " +
+                                "classifierUpdate should be either a %s or a %s but found %s",
+                        classIndex, Attribute.Type.BOOL, Attribute
+                                .Type.STRING, classLabelAttributeType));
+            }
+
+        } else {
+            throw new SiddhiAppValidationException((classIndex) + "th parameter is not " +
+                    "an attribute (VariableExpressionExecutor) present in the stream definition. Found a "
+                    + attributeExpressionExecutors[classIndex].getClass().getCanonicalName());
+        }
+        return classLabelVariableExecutor;
+    }
+
+    /**
+     * @param model
+     * @param noOfFeatures
+     * @return
```

Comment:
Same applies here as well.

Reply:
Fixed in [#d65bc3a9237b1178ce3d2d50ac0fab55709e2e0a](https://github.com/wso2-extensions/siddhi-gpl-execution-streamingml/pull/16/commits/d65bc3a9237b1178ce3d2d50ac0fab55709e2e0a)